services/merak-compute/handler/create.go

```diff
@@ -144,22 +132,31 @@ func caseCreate(ctx context.Context, in *pb.InternalComputeConfigInfo) (*pb.Retu
 		// Execute VM creation on a per pod basis
 		// Send a list of VMs to the Workflow
 		workflowOptions = client.StartWorkflowOptions{
-			ID:          common.VM_CREATE_WORKFLOW_ID + strconv.Itoa(n),
-			TaskQueue:   common.VM_TASK_QUEUE,
-			RetryPolicy: retrypolicy,
+			ID:                       common.VM_CREATE_WORKFLOW_ID + strconv.Itoa(n),
+			TaskQueue:                common.VM_TASK_QUEUE,
+			RetryPolicy:              retrypolicy,
+			WorkflowExecutionTimeout: common.TEMPORAL_WF_EXEC_TIMEOUT,
+			WorkflowRunTimeout:       common.TEMPORAL_WF_RUN_TIMEOUT,
+			WorkflowTaskTimeout:      common.TEMPORAL_WF_TASK_TIMEOUT,
 		}
-		log.Println("Executing VM Create Workflow with VMs ", vms.Val())
-		we, err := TemporalClient.ExecuteWorkflow(context.Background(), workflowOptions, create.Create, vms.Val())
+		num_vms := strconv.Itoa(len(vms.Val()))
+		count += len(vms.Val())
+		log.Println("Executing VM Create Workflow with VMs " + num_vms + " on pod at " + pod.ContainerIp)
+		_, err := TemporalClient.ExecuteWorkflow(context.Background(), workflowOptions, create.Create, vms.Val(), pod.ContainerIp)
 		if err != nil {
 			return &pb.ReturnComputeMessage{
 				ReturnMessage: "Unable to execute create workflow",
 				ReturnCode:    commonPB.ReturnCode_FAILED,
 				Vms:           returnVMs,
 			}, err
 		}
-		log.Println("Started Create workflow WorkflowID "+we.GetID()+" RunID ", we.GetRunID())
 	}
 
+	returnVM := pb.InternalVMInfo{
+		Id: "Started deployment for " + strconv.Itoa(count) + " VMs",
+	}
+	returnVMs = append(returnVMs, &returnVM)
+	log.Println("Started deployment for " + strconv.Itoa(count) + " VMs")
```

Comment:
@phudtran we may need to redesign the return message for VM info. Roughly ideas include summary and detailed return. For summary, you can return the number of VMs in each status and return VM IDs for the error or ongoing VM. For the detailed return, we can base on the VM IDs to return the detailed VM info. What do you think?

Reply:
Sure that sounds like a good option to have for the INFO case.